### ISSUE TITLE
[Bug]: [Nightly]Search failed because fail to get shard leaders from QueryCoord: a channel is not available in any replica

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#21000]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 039e9ce
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus2.3.0.dev15
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed because fail to get shard leaders from QueryCoord: a channel is not available in any replica.
```
[2022-12-05T19:25:31.805Z] [2022-12-05 18:28:07 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=All attempts results:
[2022-12-05T19:25:31.805Z] attempt #1:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #2:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #3:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #4:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #5:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #6:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #7:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #8:context deadline exceeded
[2022-12-05T19:25:31.805Z] )>, <Time:{'RPC start': '2022-12-05 18:27:54.176991', 'RPC error': '2022-12-05 18:28:07.279683'}> (decorators.py:108)
[2022-12-05T19:25:31.805Z] [2022-12-05 18:28:07 - ERROR - ci_test]: Traceback (most recent call last):
[2022-12-05T19:25:31.805Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2022-12-05T19:25:31.805Z]     res = func(*args, **_kwargs)
[2022-12-05T19:25:31.805Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2022-12-05T19:25:31.805Z]     return func(*arg, **kwargs)
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 610, in search
[2022-12-05T19:25:31.805Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2022-12-05T19:25:31.805Z]     raise e
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2022-12-05T19:25:31.805Z]     return func(*args, **kwargs)
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2022-12-05T19:25:31.805Z]     ret = func(self, *args, **kwargs)
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2022-12-05T19:25:31.805Z]     raise e
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2022-12-05T19:25:31.805Z]     return func(self, *args, **kwargs)
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 469, in search
[2022-12-05T19:25:31.805Z]     return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 438, in _execute_search_requests
[2022-12-05T19:25:31.805Z]     raise pre_err
[2022-12-05T19:25:31.805Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 429, in _execute_search_requests
[2022-12-05T19:25:31.805Z]     raise MilvusException(response.status.error_code, response.status.reason)
[2022-12-05T19:25:31.805Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=All attempts results:
[2022-12-05T19:25:31.805Z] attempt #1:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #2:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #3:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #4:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #5:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #6:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #7:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #8:context deadline exceeded
[2022-12-05T19:25:31.805Z] )>
[2022-12-05T19:25:31.805Z]  (api_request.py:39)
[2022-12-05T19:25:31.805Z] [2022-12-05 18:28:07 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=All attempts results:
[2022-12-05T19:25:31.805Z] attempt #1:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_437849404045288691v0 is not available in any replica
[2022-12-05T19:25:31.805Z] attempt #2:fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_202_4378494040452886...... (api_request.py:40)
```

### Expected Behavior

pass

### Steps To Reproduce

```markdown
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/211/pipeline/143/
```

### Milvus Log

1. log: [artifacts-milvus-distributed-kafka-nightly-211-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/211/artifact/artifacts-milvus-distributed-kafka-nightly-211-pymilvus-e2e-logs.tar.gz)
2. collection name: search_collection_0nq0UY4n
3. failed timeline: [2022-12-05T18:28:07.430Z] [gw0] [ 60%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_with_expression[32-False-int64 == 0 || int64 == 1 || int64 == 2]

**Comment by yah01** [comment]
![image](https://user-images.githubusercontent.com/12216890/205841310-e7084937-b219-415b-a1bc-bff3731e59cb.png)
QueryNode panic occurs while loading indexed field

**Comment by yah01** [comment]
![image](https://user-images.githubusercontent.com/12216890/205869586-01fc00c9-6f90-45f7-b77b-6838b017abca.png)

As this is a known issue in the engine layer
/assign @hhy3

**Comment by NicoYuan1986** [comment]
Latest: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/212/pipeline/187
case: TestLoadCollection.test_load_replica_partitions

**Comment by moliqingwa** [comment]
How to avoid the issue, because we have this issue too?

**Comment by yah01** [comment]
> How to avoid the issue, because we have this issue too?

Do you use DiskANN?

**Comment by moliqingwa** [comment]
> > How to avoid the issue, because we have this issue too?
> 
> Do you use DiskANN?

No

**Comment by yah01** [comment]
The `failed to get shard leaders` issues have been fixed with upgrading Knowhere and #21069 
/assign @NicoYuan1986 
@moliqingwa If you meet this issue in 2.2.0, please create a issue and post your logs then we will help. As I know, this issue should only occur for master branch

**Comment by NicoYuan1986** [comment]
The error hasn't reproduced these days.
